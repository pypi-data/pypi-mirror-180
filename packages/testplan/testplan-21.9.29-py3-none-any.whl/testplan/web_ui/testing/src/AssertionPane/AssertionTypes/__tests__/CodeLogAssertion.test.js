import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import CodeLogAssertion from '../CodeLogAssertion';

function defaultProps() {
  return {
    assertion: {
      "category": "DEFAULT",
      "machine_time": "2019-02-12T17:41:43.302500+00:00",
      "description": null,
      "language": "c",
      "code": '#include <stdio.h>\nint main()\n{\n    printf("Hello world!")\n    return 0;\n}\n',
      "line_no": 123,
      "meta_type": "entry",
      "type": "CodeLog",
      "utc_time": "2019-02-12T17:41:43.302494+00:00"
    }
  };
}


describe('CodeLogAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<CodeLogAssertion {...props} />);
    expect(shallowComponent).toMatchSnapshot();
  });
});
