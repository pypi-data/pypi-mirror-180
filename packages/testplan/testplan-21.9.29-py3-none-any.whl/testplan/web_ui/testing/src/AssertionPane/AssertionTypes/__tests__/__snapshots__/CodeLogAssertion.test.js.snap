// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeLogAssertion shallow renders the correct HTML structure 1`] = `
<SyntaxHighlighter
  language="c"
  showLineNumbers={true}
>
  #include &lt;stdio.h&gt;
int main()
{
    printf("Hello world!")
    return 0;
}

</SyntaxHighlighter>
`;
